src/conn/conn_reconfig.c

```diff
@@ -13,315 +13,272 @@
+            unchg = true;
+        else {
+            /*
+             * We're doing an upgrade or downgrade, check whether
+             * transactions are active.
+             */
+            WT_RET(__wt_txn_activity_check(session, &txn_active));
+            if (txn_active)
+                WT_RET_MSG(session, ENOTSUP, "system must be quiescent"
+                                             " for upgrade or downgrade");
+        }
+        F_SET(conn, WT_CONN_COMPATIBILITY);
+    }
+    /*
+     * If we're a reconfigure and the user did not set any compatibility
+     * or did not change the setting, we're done.
+     */
+    if (reconfig && (!F_ISSET(conn, WT_CONN_COMPATIBILITY) || unchg))
+        goto done;
+
+    /*
+     * The maximum and minimum required version for existing files
+     * is only available on opening the connection, not reconfigure.
+     */
+    WT_RET(__wt_config_gets(session, cfg, "compatibility.require_min", &cval));
+    if (cval.len != 0)
+        WT_RET(__conn_compat_parse(session, &cval, &min_major, &min_minor));
+
+    WT_RET(__wt_config_gets(session, cfg, "compatibility.require_max", &cval));
+    if (cval.len != 0)
+        WT_RET(__conn_compat_parse(session, &cval, &max_major, &max_minor));
+
+    /*
+     * The maximum required must be greater than or equal to the
+     * compatibility release we're using now. This is on an open and we're
+     * checking the two against each other. We'll check against what was
+     * saved on a restart later.
+     */
+    if (!reconfig && max_major != WT_CONN_COMPAT_NONE &&
+      (max_major < rel_major || (max_major == rel_major && max_minor < rel_minor)))
+        WT_RET_MSG(session, ENOTSUP,
+          WT_COMPAT_MSG_PREFIX "required max of %" PRIu16 ".%" PRIu16
+                               "cannot be smaller than compatibility release %" PRIu16 ".%" PRIu16,
+          max_major, max_minor, rel_major, rel_minor);
+
+    /*
+     * The minimum required must be less than or equal to the compatibility
+     * release we're using now. This is on an open and we're checking the
+     * two against each other. We'll check against what was saved on a
+     * restart later.
+     */
+    if (!reconfig && min_major != WT_CONN_COMPAT_NONE &&
+      (min_major > rel_major || (min_major == rel_major && min_minor > rel_minor)))
+        WT_RET_MSG(session, ENOTSUP,
+          WT_COMPAT_MSG_PREFIX "required min of %" PRIu16 ".%" PRIu16
+                               "cannot be larger than compatibility release %" PRIu16 ".%" PRIu16,
+          min_major, min_minor, rel_major, rel_minor);
+
+    /*
+     * On a reconfigure, check the new release version against any
+     * required maximum version set on open.
+     */
+    if (reconfig && conn->req_max_major != WT_CONN_COMPAT_NONE &&
+      (conn->req_max_major < rel_major ||
+          (conn->req_max_major == rel_major && conn->req_max_minor < rel_minor)))
```

Comment:
More weird indentation

Reply:
As discussed, it's indented even further to avoid aligning with the body of the `if` statement.